File: HotelBooking123 copy/src/main/java/com/example/demo/service/HotelService.java
package com.example.demo.service;

import java.util.List;
import com.example.demo.model.Hotel;

public interface HotelService {
	public Hotel addHotel(Hotel hotel);
	public List<Hotel> getAllHotels();
	public Hotel getHotelById(int hotelId);
	public Hotel updateHotel(int hotelId,Hotel hotel);
	public List<Hotel> deleteHotel(int hotelId);
}

File: src/app/src/app/hotel.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class HotelService {
  url = "http://localhost:8085/api/hotel/"
  constructor(private httpClient: HttpClient) { }
  getAllHotel() {
    return this.httpClient.get(`${this.url}`);
  }

  hotelSignUp(hotel: any) {
    return this.httpClient.post(`${this.url}`, hotel);
  }
  addHotel(hotel: any) {
    return this.httpClient.post(this.url, hotel);
  }

  getHotelById(hotelId: number) {
    return this.httpClient.get(`${this.url}${hotelId}`);
  }

  updateHotel(hotelId: number, hotel: any) {
    return this.httpClient.put(`${this.url}${hotelId}`, hotel);
  }

  deleteHotelById(hotelId: number) {
    return this.httpClient.delete(`${this.url}${hotelId}`);
  }
}